x/cosmwasmpool/genesis_test.go

```diff
@@ -0,0 +1,151 @@
+package cosmwasmpool_test
+
+import (
+	"fmt"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v16/x/cosmwasmpool/types"
+
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+)
+
+func (s *PoolModuleSuite) TestInitGenesis() {
+	s.Setup()
+
+	expectedTotalLiquidity := sdk.Coins{}
+	for i := 0; i < 3; i++ {
+		s.FundAcc(s.TestAccs[0], initalDefaultSupply)
+		pool := s.PrepareCustomTransmuterPool(s.TestAccs[0], defaultDenoms)
+		s.JoinTransmuterPool(s.TestAccs[0], pool.GetId(), initalDefaultSupply)
+		expectedTotalLiquidity = expectedTotalLiquidity.Add(initalDefaultSupply...)
+		fmt.Println("poolsss", pool.GetTotalPoolLiquidity(s.Ctx))
+	}
+
+	pool, _ := s.App.CosmwasmPoolKeeper.GetPoolById(s.Ctx, 1)
+	fmt.Println("pool liq", pool.GetTotalPoolLiquidity(s.Ctx))
+
+	pools, err := s.App.CosmwasmPoolKeeper.GetPools(s.Ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	cosmwasmPool, _ := pools[0].(types.CosmWasmExtension)
+	fmt.Println("cosmwasmPool", cosmwasmPool.GetTotalPoolLiquidity(s.Ctx))
+
+	cosmwasmPoolPreInit := pools[0]
+
+	sf := cosmwasmPoolPreInit.GetSpreadFactor(s.Ctx)
+	fmt.Println("spread factor", sf)
+
+	poolAnys := []*codectypes.Any{}
+	for _, poolI := range pools {
+		cosmwasmPool, ok := poolI.(types.CosmWasmExtension)
+		if !ok {
+			panic("invalid pool type")
+		}
+		sf := cosmwasmPool.GetSpreadFactor(s.Ctx)
+		fmt.Println("spread factor 2", sf)
+		cosmwasmPool.SetWasmKeeper(s.App.WasmKeeper)
+		any, err := codectypes.NewAnyWithValue(cosmwasmPool)
+		if err != nil {
+			panic(err)
+		}
+		poolAnys = append(poolAnys, any)
+	}
+
+	fmt.Println("pools liq", pools[0].(types.CosmWasmExtension).GetTotalPoolLiquidity(s.Ctx))
+
+	liquiditytotal, err := s.App.CosmwasmPoolKeeper.GetTotalLiquidity(s.Ctx)
+	s.Require().NoError(err)
+	fmt.Println("TOTAL PRE liquidity", liquiditytotal.String())
+
+	// We need to export the wasm module and reimport it after we reset the test environment
+	// This is because cosmwasmpools point to a contract address, and if we don't export this as well
+	// this test will fail.
```

Comment:
Why? I don't see why we need to export / import `model.Pool` that has the `wasmKeeper`

Reply:
Moving this over from DMs:

Without export / import genesis (and calling the GetPoolById) I still get the error when trying to GetSpreadFactor:

```
--- FAIL: TestPoolModuleSuite/TestInitGenesis (0.24s)
       /osmosis/x/cosmwasmpool/suite.go:87: test panicked: contract: not found [osmosis-labs/wasmd@v0.31.0-osmo-v16/x/wasm/keeper/keeper.go:714]
            goroutine 10 [running]:
            runtime/debug.Stack()
```

Reply:
Discussed offline. This makes sense to me now